experiments/compositions/composition/internal/controller/expander_reconciler.go

```diff
@@ -324,6 +331,16 @@ func (r *ExpanderReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		newStatus.ClearCondition(compositionv1alpha1.Ready)
 		message := fmt.Sprintf("Evaluated stages: %s \n Applied stages: %s", strings.Join(stagesEvaluated, ", "), strings.Join(stagesApplied, ", "))
 		newStatus.AppendCondition(compositionv1alpha1.Ready, metav1.ConditionFalse, message, "PendingStages")
+
+		if expanderLogsEnabled {
+			// TODO(@xiaoweim): Log ressource UID in the future if possible
+			r.Recorder.Event(&inputcr, "Normal", fmt.Sprintf("Finished expander stage %d: %s", index, expander), fmt.Sprintf("version: %d, resource count: %d", inputcr.GetGeneration(), applier.Count()))
+			for i, resourceStatus := range newStatus.Stages[expander].LastApplied {
+				logger.Info("Resource %d\n", i)
+				logger.Info("Resource name: %s, resource namespace: %s, resource gvk: %s %s %s, resource status: %s",
```

Comment:
I could be wrong. But i think @cheftako's intent was to include the whole object here. 
I would do it evaluateAndSavePlan() where we call the GRPC interface.

The goal is to record the GRPC inputs and response.